### PR TITLE
#1097 Allow to pass driver specific options from configuration

### LINKED ISSUE #676: Some of Configuration.browserCapabilities are lost
Are not added capabilities chrome
Selenide: 4.10

Examle code:
```
@BeforeSuite
public void setUp() {
  ChromeOptions chromeOptions = new ChromeOptions();
  chromeOptions.addArguments(("load-extension=C:\Users\extension"));
  chromeOptions.addExtensions(new File("/path/to/extension.crx"));
  Configuration.browserCapabilities = new DesiredCapabilities();
  Configuration.browserCapabilities.setCapability(ChromeOptions.CAPABILITY, chromeOptions);
  Configuration.browser = (getPropertyConfig("browser"));
}
```

@BorisOsipov 11:09:
@VadimGulich да проблема есть, сделаешь ишью на гитхабе?

@VadimGulich 11:09: 
@BorisOsipov Да, сделаю

### LINKED ISSUE #1097: Allow to pass driver specific options from configuration
## The problem

As Selenide user I want to pass `ChromeOptions` \ `FirefoxOptions` \ `EtcOptions` from `Configuration` without `WebDriverProvider`

Now it is not possible due bug https://github.com/selenide/selenide/issues/676 (actually selenium capabilities merge bug) - some options are lost on capabilities merging.

### PR DESCRIPTION
**Pull request by BorisOsipov**
## Proposed changes
Allow to change `*Options` rigth before Selenide creates driver by driver factories.
Resolve #1097 Allow to pass driver specific options from configuration
Fix #676

## Checklist
- [ ] Checkstyle and unit tests are passed locally with my changes by running `gradlew check chrome_headless firefox_headless` command
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: statics/src/test/java/integration/BrowserOptionsInterceptorSampleUsageTest.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package integration;
+
+import com.codeborne.selenide.Configuration;
+import org.junit.jupiter.api.Test;
+
+class BrowserOptionsInterceptorSampleUsageTest extends IntegrationTest {
+  @Test
+  void canModifyOptionsByConfig() {
+    Configuration.browserOptionsInterceptors.chromeOptionsInterceptor = options -> {
```
Comment (dstekanov) [inline statics/src/test/java/integration/BrowserOptionsInterceptorSampleUsageTest.java:9]: Suggestion for naming: 
`Configuration.extraBrowserOptions.chrome` or `Configuration.additionalBrowserOptions.chrome`

Comment (dstekanov) [reply statics/src/test/java/integration/BrowserOptionsInterceptorSampleUsageTest.java:9]: Seems it not just add extra options but also can override it...

**Comment by asolntsev** [comment]
@BorisOsipov Probably the problem could be easily solvable if `ChromeDriverFactory` implements `WebDriverProvider`? Then everyone can inherit it and override only one small method which sets chrome options. I will try such approach.

**Comment by asolntsev** [comment]
Probably superseded by https://github.com/selenide/selenide/pull/1155

**Pull request closed without merging**